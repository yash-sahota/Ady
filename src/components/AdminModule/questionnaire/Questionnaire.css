.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.Questionnaire-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: #007bff;
}
p{
  color: black;
}
.Questionnaire-Area{
  margin-left: 10%;
  margin-right: 5%;
}

label{
  display: inline !important;
  padding-left: 2%;
}

.Questionnaire-Heading{
  border-radius: 25px;
  background: #007bff;
  padding-left: 1.5%;
  padding-top: 1%;
  padding-bottom: 1%;
  color: white;
}


.Questionnaire-Answer-Area{
  margin-left: 2%;
}

.Questionnaire-Continue-Button{
  border-radius: 25px !important;
  width: 100%;
  margin-top: 5%;
  padding: 1% !important;
}

.App-link {
  color: #007bff;
}

.modal-content{
  width: 150%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

