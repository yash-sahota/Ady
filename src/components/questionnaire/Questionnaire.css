:root {
  --primary: #00d0cc;
  --primarysecond: #f6ceb4;
  --secondary: #FF6563;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.Questionnaire-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: #00D0CC;
}
p{
  color: black;
}
.Questionnaire-Area{
  border: 0;
  border-radius: 25px;
  margin-left: 5%;
  margin-right: 5%;
  /* padding-left: 10px; */
  height: 80vh;
  overflow-y: scroll;
  overflow-x: hidden;
  background: #EEE9DF;
}

label{
  display: inline !important;
  padding-left: 2%;
}

.Questionnaire-Heading{

  text-align: left;
  color: var(--primary);
}


.Questionnaire-Answer-Area{
  margin-left: 3%;
}

.Questionnaire-subtopic-Area{
  margin-left: 2%;
}

.Questionnaire-Action-Area-padding{
  padding-left: 30% !important;
}

.Questionnaire-Continue-Button{
  border-radius: 25px !important;
  width: 100%;
  margin-top: 5%;
  padding: 1% !important;
}

.Questionnaire-Question-Panel:not(:last-child){
  /*padding: 10px;*/
  /* border-bottom: 1px solid; */
  /* border-radius: 25px; */
  box-shadow: 0px 3.794px 13px 0px #f1f1f1;
  margin-bottom: 1%;
}

.Questionnaire-Edit-Div-Border{
  border-radius: 20px !important;
  border:1px solid black;
}

.Questionnaire-Next-Back_Button{
  margin-left: 40%;
}


.App-link {
  color: #007bff;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.modal-content{
  width: 150%;
}

.Questionnaire-Question-Panel {
  padding: 20px;
  /*min-height: 360px;*/
  border-radius: 22px;
  background: #F1E8DB;
}

#questionnaire-view .Questionnaire-Question-Panel {
  /*height: 400px;
  overflow-y: scroll;*/
  margin-bottom: 10px;
}

.questionnaaire-container {
  height: 620px;
  overflow-y: scroll;
}

/* width */

#questionnaire-view .Questionnaire-Question-Panel::-webkit-scrollbar, 
.Questionnaire-Area::-webkit-scrollbar,
.questionnaaire-container::-webkit-scrollbar {
  width: 5px;
}

/* Track */

#questionnaire-view .Questionnaire-Question-Panel::-webkit-scrollbar-track, 
.Questionnaire-Area::-webkit-scrollbar-track,
.questionnaaire-container::-webkit-scrollbar-track {
  box-shadow: inset 0 0 15px #fff;
  border-radius: 10px;
}

/* Handle */

#questionnaire-view .Questionnaire-Question-Panel::-webkit-scrollbar-thumb, 
.Questionnaire-Area::-webkit-scrollbar-thumb,
.questionnaaire-container::-webkit-scrollbar-thumb {
  background: var(--primary);
  border-radius: 10px;
}

/* Handle on hover */

#questionnaire-view .Questionnaire-Question-Panel::-webkit-scrollbar-thumb:hover, 
.Questionnaire-Area::-webkit-scrollbar-thumb:hover,
.questionnaaire-container::-webkit-scrollbar-thumb:hover {
  background: var(--primary);
}