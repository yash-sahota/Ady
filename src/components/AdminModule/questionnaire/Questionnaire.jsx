import React from 'react';
import '../../../component/questionnaire/Questionnaire.css';
import {Button} from 'react-bootstrap';
import {saveQuestionAnswer} from "../../../component/questionnaire/QuestionAnswerService";
import {getQuestionnaires} from "../../../component/questionnaire/QuestionnaireService";
import article from '../../../images/articles@3x.png';
import Header from "../../Login Module/Header";
import Footer from "../../Login Module/Footer";

class Questionnaire extends React.Component {


    state = {
        isLoading: true,
        questionnaire: null,
        error: null,
        currentPanel: 0
    }


    async componentDidMount() {
        // GET request using fetch with async/await
        const response = await getQuestionnaires();
        this.setState({questionnaire: response, isLoading: false})
    }

    render() {

        const {isLoading, questionnaire, error, currentPanel} = this.state;
        var topicSet = new Set()
        return (
            <div>
                <Header />
                <div className="row">
                    <div className="col-md-6">
                        <img
                            src={article}
                            alt="HealthierU Logo"
                            style={{width: "100%"}}
                        />
                    </div>
                    <div className="col-md-6">
                        <br/>
                        <br/>
                        <div className="Questionnaire-header">
                            <h1>Health Behaviours</h1>
                            <p> Do You Suffer from Any of the Following?</p>
                        </div>
                        {!isLoading ? (
                            this.state.questionnaire.map((item, index) =>

                                item.published ? <div className="Questionnaire-Area" key={index}>
                                    {Object.entries(item.topicSubtopicDetails).map((dataItem, subIndex) => {

                                        //console.log(Object.entries(item.topicSubtopicDetails).length)
                                        topicSet.add(dataItem[0].split("#")[0])
                                        return <div hidden={currentPanel !== subIndex}>
                                            <div className={"Questionnaire-Question-Panel"}>

                                                <div className="Questionnaire-Heading">
                                                   <h4> {dataItem[0].split("#")[0]}
                                                   </h4>
                                                </div>


                                                <div className="Questionnaire-subtopic-Area">
                                                    <b>{dataItem[0].split("#").length > 2 ? dataItem[0].split("#")[1] : ""}</b>
                                                </div>

                                                {dataItem[1].map((question) => {
                                                    return <div className="Questionnaire-Answer-Area">
                                                        <div hidden={question.questiontype	=== 'TEXT'}>
                                                            <input type="checkbox"
                                                                   className="form-radio"
                                                                   onChange={this.handleCheckBoxEvent}
                                                                   id={question.id}/>
                                                            <label id="label_input_6_0">{question.question}</label>
                                                        </div>

                                                        <div hidden={question.questiontype	=== 'BOOLEAN'}>
                                                            <div className="form-group row">
                                                                <label htmlFor="description" className="col-sm-6 col-form-label">{question.question}</label>
                                                                <div className="col-sm-11">
                                                                    <input type="text"  className="form-control"
                                                                           id={question.id}
                                                                           placeholder={question.question}  onBlur={this.handleTextBoxEvent}></input>
                                                                </div>
                                                            </div>
                                                        </div>
                                                    </div>
                                                })}
                                            </div>

                                            <div  className={"Questionnaire-Next-Back_Button"} >
                                                
                                                <Button  className="btn btn-primary mr-2" hidden={ 0 === currentPanel}
                                                        onClick={(e) => this.backPanel(e)}>Back</Button>
                                                <Button className="btn btn-primary mr-2" hidden={ Object.entries(item.topicSubtopicDetails).length -1 === currentPanel}
                                                        onClick={(e) => this.nextPanel(e)}>Next</Button>
                                            </div>
                                            <br/>
                                        </div>


                                    })}


                                </div> : <div></div>
                            )
                            // If there is a delay in data, let's let the user know it's loading
                        ) : (
                            <h3>Loading...</h3>
                        )}

                        {!isLoading ? (
                            <div className="Questionnaire-Area">
                                <Button type="submit" variant="primary" className="Questionnaire-Continue-Button"
                                        onClick={(e) => this.create(e)}>Continue</Button>
                            </div>
                        ) : <div></div>}
                    </div>
                </div>
                <Footer />

            </div>

        );
    }


    async handleTextBoxEvent(event) {
        //console.log("Event Checkbox ");
        //console.log(event.target.id)
        //console.log(event.target.value)

        var data = {
            "answer": event.target.value,
            "patientId": null,
            "questionId": event.target.id,
            "id": event.target.name === "" ? null : event.target.name
        };

        //console.log(await saveQuestionAnswer(data));
    }


    async handleCheckBoxEvent(event) {
        //console.log("Event Checkbox ");
        //console.log(event.target.checked)
        //console.log(event.target.id)

        //console.log(event)
        var data = {
            "answer": event.target.checked,
            "patientId": null,
            "questionId": event.target.id,
            "id": event.target.name === "" ? null : event.target.name
        };


        //console.log(await saveQuestionAnswer(data));
    }



    nextPanel(event) {
        //console.log(this.state.currentPanel)
        this.setState({
            currentPanel: this.state.currentPanel + 1
        })
    }
    backPanel(event) {
        //console.log(this.state.currentPanel)
        this.setState({
            currentPanel: this.state.currentPanel - 1
        })
    }
}

export default Questionnaire;