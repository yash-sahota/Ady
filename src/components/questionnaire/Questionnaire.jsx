import React from 'react';
import './Questionnaire.css';
import {Button} from 'react-bootstrap';
import {getQuestionnaires} from "./QuestionnaireService";
import {saveQuestionAnswer} from "./QuestionAnswerService";
import TransparentLoader from '../Loader/transparentloader';

class Questionnaire extends React.Component {


    state = {
        isLoading: true,
        questionnaire: null,
        error: null
    }


    async componentDidMount() {
        // GET request using fetch with async/await
        const response = await getQuestionnaires();
        if(response && response.status === 200) {
            this.setState({questionnaire: response, isLoading: false})
        }
        
    }

    render() {

        const {isLoading, questionnaire, error} = this.state;
        return (
            <div>
                {isLoading && (
                    <TransparentLoader />
                )}
                <div className="Questionnaire-header">
                    <h1>Health Behaviuor</h1>
                    <p> Do you suffer from any of following ?</p>
                </div>


                {questionnaire && (
                    questionnaire.map((item, index) =>

                        item.published ? <div className="Questionnaire-Area" key={index}>
                            <label
                                id="label_6">
                                <div className="Questionnaire-Heading">
                                    {item.description}
                                </div>
                            </label>
                            <div data-layout="full" className="Questionnaire-Answer-Area">
                                <div role="group" aria-labelledby="label_6" data-component="radio">
                                    {this.state.questionnaire[index].questions.map((question, subIndex) =>
                                        <div>
                                            <input type="checkbox" tabIndex={subIndex} className="form-radio"
                                                   onChange={this.handleCheckBoxEvent}
                                                   id={question.id}/>
                                            <label
                                                id="label_input_6_0">{question.question}</label>
                                        </div>
                                    )}

                                </div>
                            </div>
                        </div> : <div></div>
                    )
                    // If there is a delay in data, let's let the user know it's loading
                )}

                
                    <div className="Questionnaire-Area-continue-button">
                        <Button type="submit" variant="primary" className="Questionnaire-Continue-Button"
                                onClick={(e) => this.create(e)}>Continue</Button>
                    </div>
                <br/>
            </div>

        );
    }

    async handleCheckBoxEvent(event) {
        //console.log("Event Checkbox ");
        //console.log(event.target.checked)
        //console.log(event.target.id)

        //console.log(event)
        var data = {
            "answer": event.target.checked,
            "patientId": null,
            "questionId": event.target.id,
            "id": event.target.name === "" ? null : event.target.name
        };


        //console.log(await saveQuestionAnswer(data));

    }
}

export default Questionnaire;