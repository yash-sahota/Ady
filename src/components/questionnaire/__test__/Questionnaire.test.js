import { shallow } from 'enzyme';
import React from 'react';
import Questionnaire from "../Questionnaire";
import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Questionnaire />', () => {

    it("should render my component", () => {
        const component  = shallow(
            <Questionnaire/>
        );

        expect(component.getElements()).toMatchSnapshot();
    });

});