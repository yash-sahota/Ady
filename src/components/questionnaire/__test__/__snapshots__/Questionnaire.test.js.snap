// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Questionnaire /> should render my component 1`] = `
Array [
  <div>
    <div
      className="Questionnaire-header"
    >
      <h1>
        Health Behaviuor
      </h1>
      <p>
         Do you suffer from any of following ?
      </p>
    </div>
    <h3>
      Loading...
    </h3>
    <div />
    <br />
  </div>,
]
`;
